mithril-signer/src/runtime/runner.rs

```diff
@@ -255,9 +264,23 @@ impl Runner for SignerRunner {
             return Ok(false);
         }
 
-        if let Some(signer) =
-            pending_certificate.get_signer(self.services.single_signer.get_party_id())
-        {
+        // TODO XXX How handle errors ? It occurs in test_create_immutable_files_full_single_signature integration test (no signers in epoch 1)
+        // TODO XXX Do we warn and return "can not sign" or return an error ?
+        let current_signer_with_stake: Option<SignerWithStake> = {
+            let epoch_service = self.services.epoch_service.read().await;
+            let current_signers = epoch_service.current_signers_with_stake().await;
```

Comment:
You could have used the newly introduced `get_current_signers_with_stake` function

Reply:
We use the new `runner` function `get_current_signers()

Reply:
Done